chisel/src/dispatcher.rs

```diff
@@ -0,0 +1,934 @@
+                    self.errored = true;
+                    DispatchResult::Failure(Some(e.to_string()))
+                }
+            }
+        }
+    }
+
+    /// Decodes traces in the [ChiselResult]
+    /// TODO: Add `known_contracts` back in.
+    ///
+    /// ### Takes
+    ///
+    /// - A reference to a [SessionSourceConfig]
+    /// - A mutable reference to a [ChiselResult]
+    ///
+    /// ### Returns
+    ///
+    /// Optionally, a [CallTraceDecoder]
+    pub fn decode_traces(
+        &self,
+        session_config: &SessionSourceConfig,
+        result: &mut ChiselResult,
+        // known_contracts: &ContractsByArtifact,
+    ) -> eyre::Result<CallTraceDecoder> {
+        let mut etherscan_identifier = EtherscanIdentifier::new(
+            &session_config.foundry_config,
+            session_config.evm_opts.get_remote_chain_id(),
+        )?;
+
+        let mut decoder =
+            CallTraceDecoderBuilder::new().with_labels(result.labeled_addresses.clone()).build();
+
+        decoder.add_signature_identifier(SignaturesIdentifier::new(
+            Config::foundry_cache_dir(),
+            session_config.foundry_config.offline,
+        )?);
+
+        for (_, trace) in &mut result.traces {
+            // decoder.identify(trace, &mut local_identifier);
+            decoder.identify(trace, &mut etherscan_identifier);
+        }
+        Ok(decoder)
+    }
+
+    /// Display the gathered traces of a REPL execution.
+    ///
+    /// ### Takes
+    ///
+    /// - A reference to a [CallTraceDecoder]
+    /// - A mutable reference to a [ChiselResult]
+    ///
+    /// ### Returns
+    ///
+    /// Optionally, a unit type signifying a successful result.
+    pub async fn show_traces(
+        &self,
+        decoder: &CallTraceDecoder,
+        result: &mut ChiselResult,
+    ) -> eyre::Result<()> {
+        if result.traces.is_empty() {
+            eyre::bail!("Unexpected error: No traces gathered. Please report this as a bug: https://github.com/foundry-rs/foundry/issues/new?assignees=&labels=T-bug&template=BUG-FORM.yml");
+        }
+
+        println!("{}", Paint::green("Traces:"));
+        for (kind, trace) in &mut result.traces {
+            // Display all Setup + Execution traces.
+            if matches!(kind, TraceKind::Setup | TraceKind::Execution) {
+                decoder.decode(trace).await;
+                println!("{trace}");
+            }
+        }
+
+        Ok(())
+    }
+
+    /// Format a type that implements [fmt::Display] as a chisel error string.
+    ///
+    /// ### Takes
+    ///
+    /// A generic type implementing the [std::fmt::Display] trait.
+    ///
+    /// ### Returns
+    ///
+    /// A formatted error [String].
+    fn make_error<T: std::fmt::Display>(msg: T) -> String {
+        format!("{} {}", Paint::red(format!("{} Chisel Error:", CHISEL_CHAR)), Paint::red(msg))
+    }
+}
+
+/// Builtin chisel command variants
+#[derive(Debug, EnumIter)]
+pub enum ChiselCommand {
```

Comment:
suggestion: move to stand alone file, or is this only used here in this file?

Reply:
This is used in multiple files- moved out to `cmd.rs`